include/vast/CodeGen/CodeGenDeclVisitor.hpp

```diff
@@ -311,9 +309,9 @@ namespace vast::cg {
         }
 
         // Implelements buildGlobalFunctionDefinition of vast codegen
-        operation build_function_prototype(clang::GlobalDecl decl, mlir_type fty) {
-            // Get or create the prototype for the function.
-            // TODO: Figure out what to do here? llvm uses a GlobalValue for the FuncOp in mlir
+        operation build_function_prototype(clang::GlobalDecl decl) {
+            const auto *fn = clang::cast< clang::FunctionDecl >(decl.getDecl());
```

Comment:
Maybe `dyn_cast` and emit error the way we usually do? (So that it can be intercepted).